facebook/ThreatExchange
# [omm] Handle indices > 2GB

Summary
---------

Phwew, this was tough. 
1. use postgres large objects to store giant indices. Since we are using a 64bit build, this gets us up to 2TB of space if I've read the docs correctly.
2. Add a ton more logging to trace various issues with timing

Test Plan
---------

1. Disable tasks in omm_config.py["pdq"]}`
2. Create a 10M PDQ hash source with settings `{"total_item_limit": 10000000, "only_signal_types": 
3. `for i in `seq 100`; do flask fetch; done`
4. flask build_indices
5. Query against a hash

See:
```
[2023-12-21 17:25:43,424] DEBUG in database: Index[pdq] importing lobject oid 202903 to tmpfile /tmp/tmpijcswiy6
[2023-12-21 17:26:05,117] DEBUG in database: Index[pdq] loaded 1625241654 bytes to tmpfile - 21 seconds
[2023-12-21 17:26:22,554] DEBUG in database: Index[pdq] deserialized - 17 seconds
[2023-12-21 17:26:22,713] DEBUG in matching: [lookup_signal] querying index
[2023-12-21 17:26:22,740] DEBUG in matching: [lookup_signal] query complete
[2023-12-21 17:26:23,837] DEBUG in matching: getting bank content
[2023-12-21 17:26:23,845] DEBUG in matching: lookup matches 36 content ids (36 enabled)
[2023-12-21 17:26:23,845] DEBUG in matching: lookup matches 3 banks (3 enabled)
```
